add rails routes

otavioschwanck opened the pull request:
### Brief summary of what the package does

Manage rails routes for Rails Development.  Go to route, add route, etc.

Ill add the README after this PR be merged.

### Direct link to the package repository

https://github.com/otavioschwanck/rails-routes

### Your association with the package

Developer

[Are you the maintainer? A contributor? An enthusiastic user?]
Contributor

### Relevant communications with the upstream package maintainer

None needed

### Checklist

<!-- Please confirm with `x`: -->

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [ ] I have confirmed some of these without doing them

<!-- After submitting, please fix any problems the CI reports. -->

purcell commented:
Hey! Thanks for submitting this. I'm a Rails users too, so this is relevant to my interests. I think it needs a bit of work to be more robust and idiomatic before wider distribution. In particular, it looks like you're quite new to Elisp and the packaging ecosystem - welcome! I sent a few easy fixes in https://github.com/otavioschwanck/rails-routes/pull/1, and feel free to ask questions about those, but I was limited for time, so here are some more notes:

- Instead of passing around values that you need to parse and split everywhere, and then look inside using `elt` with magic numbers, just make a function that reads all the routes and makes a structure for each with proper keys (e.g. controller, action etc.). [ALISTs](https://www.gnu.org/software/emacs/manual/html_node/elisp/Association-Lists.html) are good for this, and they are very convenient to access if you use the `let-alist` package. Such a function would also be more widely useful.
- Some of the function and variable naming is very confusing, sometimes because of native languages, but often just because the names and docstrings aren't accurate: it's good to pay more attention to this because the code will be very hard for others to understand and contribute to, which is part of the purpose of sharing it as a package. [Here's a great example](https://github.com/otavioschwanck/rails-routes/blob/e4691f9195d13d2f1ecd75efedd0dbf63d265352/rails-routes.el#L152), where the function does some extra stuff (substring) that is completely unexpected.
- You're depending on `projectile` anyway, so maybe just enable caching unconditionally. It will simplify the code.

Hope that helps!

otavioschwanck commented:
> Hey! Thanks for submitting this. I'm a Rails users too, so this is relevant to my interests. I think it needs a bit of work to be more robust and idiomatic before wider distribution. In particular, it looks like you're quite new to Elisp and the packaging ecosystem - welcome! I sent a few easy fixes in [otavioschwanck/rails-routes#1](https://github.com/otavioschwanck/rails-routes/pull/1), and feel free to ask questions about those, but I was limited for time, so here are some more notes:
> 
> * Instead of passing around values that you need to parse and split everywhere, and then look inside using `elt` with magic numbers, just make a function that reads all the routes and makes a structure for each with proper keys (e.g. controller, action etc.). [ALISTs](https://www.gnu.org/software/emacs/manual/html_node/elisp/Association-Lists.html) are good for this, and they are very convenient to access if you use the `let-alist` package. Such a function would also be more widely useful.
> * Some of the function and variable naming is very confusing, sometimes because of native languages, but often just because the names and docstrings aren't accurate: it's good to pay more attention to this because the code will be very hard for others to understand and contribute to, which is part of the purpose of sharing it as a package. [Here's a great example](https://github.com/otavioschwanck/rails-routes/blob/e4691f9195d13d2f1ecd75efedd0dbf63d265352/rails-routes.el#L152), where the function does some extra stuff (substring) that is completely unexpected.
> * You're depending on `projectile` anyway, so maybe just enable caching unconditionally. It will simplify the code.
> 
> Hope that helps!

Thanks for your help @purcell

About your PR, i added some comments, take a look.

About the changes that you requested, for now, i do not have time to make this changes, i did this plugin when i was on vacation and had free time.  Maybe in the future.

riscy commented on recipes/rails-routes:
```diff
@@ -0,0 +1,3 @@
+(rails-routes :repo "otavioschwanck/rails-routes"
+              :fetcher github
+              :files ("rails-routes.el"))
```
This looks like a candidate for the default recipe:
```suggestion
(rails-routes :repo "otavioschwanck/rails-routes" :fetcher github)
```

riscy reviewed (commented):
> About the changes that you requested, for now, i do not have time to make this changes, i did this plugin when i was on vacation and had free time. Maybe in the future.

This is reasonable, and there's also not a huge rush.  So I would say take your time, and we can leave the PR open for a while if you have any followup questions.

I'll also add to the above that simply loading a package should cause few side effects.  Most of these should happen as a result of the user explicitly calling a function or toggling a mode off or on.  In particular adding hooks and advice should probably be done as part of a [minor mode](https://www.gnu.org/software/emacs/manual/html_node/elisp/Minor-Modes.html).  And one last point is that I notice your repository does not have a LICENSE file which will help common tooling to detect it, which I'd recommend regardless of whether your package is indexed in MELPA.

Silex commented:
Juste mentionning https://github.com/asok/projectile-rails, maybe it could be worth considering including rails-routes into it instead of having a separate package.

otavioschwanck commented:
> Juste mentionning https://github.com/asok/projectile-rails, maybe it could be worth considering including rails-routes into it instead of having a separate package.

Projectile is more a navigation package with different approach.

riscy commented:
Hi @otavioschwanck I'll mark this one awaiting upstream for now.  Feel free to reach out.

otavioschwanck commented:
> Hi @otavioschwanck I'll mark this one awaiting upstream for now. Feel free to reach out.

@riscy feel free to merge.

riscy commented:
Apologies for letting this one slip -- I'll look again soon.

riscy commented:
Hi again @otavioschwanck, I'll leave this one awaiting upstream until you get the time to address the comments purcell and I made above.  It's understandable if you don't have time to immediately address them, but we'll likely hold off on merging until then.

otavioschwanck commented:
> Hi again @otavioschwanck, I'll leave this one awaiting upstream until you get the time to address the comments purcell and I made above. It's understandable if you don't have time to immediately address them, but we'll likely hold off on merging until then.

Hi Riscy,

I probably will not change this plugin anymore, its working very well for my usage (and my co-workers) and i just wanted to share.  If its not ready to be a plugin on melpa, i'm sorry.  If i some time in the future, i will work on a new plugin (for i18n on rails) instead of improving this plugin code.  

If you want to merge on MELPA or close this PR, feel free

Thanks!

riscy commented:
Hi @otavioschwanck -- that's reasonable, and thanks for your understanding.  I'll close this for now.
